packages/shared/test/shared-secret-compatibility.test.ts

```diff
@@ -0,0 +1,271 @@
+
+      try {
+        // Encrypt with fixed nonce
+        const ciphertext = encryptWithSharedSecret(sharedSecret, plaintext);
+
+        // Check against expected ciphertext from the JSON file
+        const ciphertextHex = Buffer.from(ciphertext).toString('hex');
+        if (expectedCiphertext) {
+          const expectedHex = Buffer.from(expectedCiphertext).toString('hex');
+          expect(ciphertextHex).toEqual(expectedHex);
+          console.debug(
+            `✓ Vector ${testVector1.id} ciphertext matches expected output`,
+          );
+        } else {
+          console.error(
+            `Generated ciphertext for vector ${testVector1.id}:`,
+            ciphertextHex,
+          );
+          console.error(
+            'Add this to the test vectors JSON file to enable validation',
+          );
+        }
+
+        // Verify decryption works correctly
+        const decrypted = decryptWithSharedSecret(sharedSecret, ciphertext);
+        expect(Buffer.from(decrypted).toString()).toEqual(
+          Buffer.from(plaintext).toString(),
+        );
+      } finally {
+        // Always restore the original implementation
+        cleanupMock();
+      }
+    });
+
+    it('should correctly encrypt and decrypt test vector 2 (empty plaintext)', () => {
+      const { sharedSecret, plaintext, fixedNonce, expectedCiphertext } =
+        testVector2;
+
+      // Set up our mock to return the fixed nonce
+      const cleanupMock = setupFixedNonceMock(fixedNonce);
+
+      try {
+        // Encrypt with fixed nonce
+        const ciphertext = encryptWithSharedSecret(sharedSecret, plaintext);
+
+        // Check against expected ciphertext from the JSON file
+        const ciphertextHex = Buffer.from(ciphertext).toString('hex');
+        if (expectedCiphertext) {
+          const expectedHex = Buffer.from(expectedCiphertext).toString('hex');
+          expect(ciphertextHex).toEqual(expectedHex);
+          console.debug(
+            `✓ Vector ${testVector2.id} ciphertext matches expected output`,
+          );
+        } else {
+          console.error(
+            `Generated ciphertext for vector ${testVector2.id}:`,
+            ciphertextHex,
+          );
+          throw Error(
+            'Add this to the test vectors JSON file to enable validation',
+          );
+        }
+
+        // Verify decryption works correctly
+        const decrypted = decryptWithSharedSecret(sharedSecret, ciphertext);
+        expect(decrypted.length).toEqual(plaintext.length);
+        expect(decrypted.byteLength).toEqual(0); // Empty plaintext
+      } finally {
+        // Always restore the original implementation
+        cleanupMock();
+      }
+    });
+  });
+
+  describe('Cross-implementation compatibility tests', () => {
+    // Compatibility test for decryption of pre-generated ciphertext from Rust
+    it('should correctly decrypt ciphertext generated by the Rust implementation', () => {
```

Comment:
Does this need to be its own `it` block, or could this simply be another structured test vector? Any reason Rust implementation needs to be separate from the other test vectors?